src/backend/services/workflow_service.py

```
@@ -396,6 +396,50 @@
 
+def _learn_from_successful_test(user_query: str, robot_code: str, test_status: str) -> None:
+    """
+    Learn from a successful test execution for pattern optimization.
+    
+    Args:
+        user_query: Original user query (None if not provided)
+        robot_code: Generated robot code
+        test_status: Test execution status
+    """
+    if test_status != 'passed':
+        logging.info(f"⏭️  Skipping pattern learning - test status: {test_status}")
```

Ruff (0.14.5)

`info()` call on root logger

(LOG015)